packages/material-ui/src/Popper/Popper.js

```diff
@@ -92,20 +96,30 @@ const Popper = React.forwardRef(function Popper(props, ref) {
       onCreate: createChainedFunction(handlePopperUpdate, popperOptions.onCreate),
       onUpdate: createChainedFunction(handlePopperUpdate, popperOptions.onUpdate),
     });
-  }, [anchorEl, disablePortal, modifiers, open, placement, placementProps, popperOptions]);
+    handlePopperRef(popper);
+  }, [
+    handlePopperRef,
+    anchorEl,
+    disablePortal,
+    modifiers,
+    open,
+    placement,
+    placementProps,
+    popperOptions,
+  ]);
 
   const handleEnter = () => {
     setExited(false);
   };
 
-  const handleClose = () => {
+  const handleClose = React.useCallback(() => {
```

Comment:
Can't see how this is tied optimization is tied to popperRef. Even more so it cascades to closing the popper whenever popperRef changes.

Reply:
The two changes are only linked by the Popper component. They can be performed independently. However, I have grouped them 👼.

Reply:
Could you describe the second change? Because it seems like a bug.

Reply:
Here?

![Capture d’écran 2019-06-05 à 16 13 03](https://user-images.githubusercontent.com/3165635/58963068-d047c580-87ac-11e9-9ea7-20a68a6b36fd.png)

Reply:
I have tried something else.